src/mongoc/mongoc-write-concern.c

```diff
@@ -380,7 +382,10 @@ bool
 _mongoc_write_concern_needs_gle (const mongoc_write_concern_t *write_concern)
 {
    if (write_concern) {
-      return ((write_concern->w != 0) && (write_concern->w != -1));
+      return (((write_concern->w != MONGOC_WRITE_CONCERN_W_UNACKNOWLEDGED) &&
+               (write_concern->w != MONGOC_WRITE_CONCERN_W_ERRORS_IGNORED)) ||
+              (write_concern->fsync_ != MONGOC_WRITE_CONCERN_FSYNC_DEFAULT && !!write_concern->fsync_) ||
+              (write_concern->journal != MONGOC_WRITE_CONCERN_JOURNAL_DEFAULT && !!write_concern->journal));
```

Comment:
These two lines duplicate logic within `mongoc_write_concern_get_fsync()` and `mongoc_write_concern_get_journal()`. Let me know if it makes sense to simply call the functions.